File: backend/src/main/java/com/example/backend/model/Enrollment.java
package com.example.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@IdClass(EnrollmentId.class)
public class Enrollment implements Serializable {

    @Id
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Id
    @ManyToOne
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    private Boolean confirmed;
    @Enumerated(EnumType.STRING)
    private EnrollmentType type;

    private int groupNumber;
}

File: frontend/src/app/model/entities/enrollment.ts
import { Course } from "./course"
import { User } from "./user"

export enum EnrollmentType
{
    MAIN_TEACHER = "MAIN_TEACHER", TEACHER = "TEACHER", STUDENT = "STUDENT"
}

export type Enrollment =
{
    id: number,
    user: User
    course: Course,
    confirmed: boolean,
    type: EnrollmentType
    groupNumber: number
}